docs/rfcs/1-add-pair-for-reacher.md

```diff
@@ -0,0 +1,64 @@
+- Author: zu1k <i@lgf.im>
+- Start Date: 2021-07-15
+- RFC PR: [beyondstorage/go-service-ipfs#13](https://github.com/beyondstorage/go-service-ipfs/pull/13)
+- Tracking Issue: [beyondstorage/go-service-ipfs#14](https://github.com/beyondstorage/go-service-ipfs/issues/14)
+
+# RFC-1: Add a pair for `Reacher`
+
+## Background
+
+With IPFS, we can get the `CID` of an object, then we can use it with a public or internal IPFS Gateway to splice out the access link.
+
+When using internal gateway like `127.0.0.1:8080`, the access link are inaccessible from external network.
+
+When using a public gateway, e.g. `ipfs.io`, `cf-ipfs.com`, if the service is an intranet IPFS cluster, it is possible that the files are not accessible from these public gateway as well.
+
+Releated issue: [beyondstorage/go-service-ipfs#5](https://github.com/beyondstorage/go-service-ipfs/issues/5)
+
+### How to get the `CID` of an object?
+
+We can [stat](https://docs.ipfs.io/reference/http/api/#api-v0-files-stat) the object, the `hash` would be its `CID`.
+
+```
+$ ipfs files stat /part1
+QmSvxBhBHn2gu9kAdjCTYfUgDNnyBcNoXXVojRyKdBjZUA  # CID
+Size: 262144
+CumulativeSize: 262158
+ChildBlocks: 0
+Type: file
+```
+
+### How to splice out the access link?
+
+- https://{gateway URL}/ipfs/{CID}/{optional path to resource}
+- https://{CID}.ipfs.{gatewayURL}/{optional path to resource}
+
+## Proposal
+
+I propose to add a pair to let the user specify the `gateway`.
```

Comment:
Can you provide more details?

- What's the gateway pair's type?
- What's the gateway pair's value? Will we use the `go-endpoint` or just plain url?
- Where can we use the pair? In `Reach` operation only?